wavefront/server/modules/knowledge_base_module/knowledge_base_module/controllers/rag_retreival_controller.py

```
@@ -96,6 +99,112 @@ def convert_uuids_to_str(data):
 
 
+def _parse_cloud_image_url(url: str) -> Tuple[str, str, str]:
+    """
+    Parse gs:// or s3:// URL into (scheme, bucket, key).
+    Returns (scheme, bucket, key) or raises ValueError.
+    """
+    url = (url or '').strip()
+    if url.startswith('gs://'):
+        rest = url[5:]
+        if '/' not in rest:
+            raise ValueError('Invalid gs:// URL: missing path after bucket')
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)